lib/posix/mqueue.c

```diff
@@ -251,7 +250,10 @@ int mq_timedsend(mqd_t mqdes, const char *msg_ptr, size_t msg_len,
 	mqueue_desc *mqd = (mqueue_desc *)mqdes;
 	s32_t  timeout;
 
-	timeout = (s32_t) timespec_to_timeoutms(abstime);
+	timeout = (s32_t) _timespec_to_timeoutms(abstime);
```

Comment:
Well, I see that this is left from the original code, but this can't be right, it just truncates s64_t to lower 32 bits, and interprets it as signed. Please do proper <= check on s64_t value. Extra points for checking if positive s64_t value won't fit into positive range of s32_t timeout, and return EINVAL in this case.

Reply:
Worse, turns out that _timespec_to_timeoutms never returns negative value, in case of negative diff it return K_NO_WAIT instead.